### ISSUE TITLE
Route refresh functionality

### ISSUE THREAD

**Issue by jaxxreal** [rackt/react-router#2038]
Hello! I'm really like using react-router for my project, but I didn't seen mention about **route refresh functionality**. 

What does it mean? I say about some functionality, which can help trigger remount current route-component. By example - I have header and footer, and route component between of this ones, and I want refresh only route-component, like a browser refresh, but without reload all page. 

I mindfully read documentation, but search nothing about my question.

**Comment by knowbody** [comment]
have a look at #1982 please.

**Comment by jaxxreal** [comment]
@knowbody thx for a link! But if I understand issue #1982 well, it's mean that I need to realize a reload method for each of my routes individually. @pdeva told about re-fetch data functionality, but I mean a centrilized, non-specific for each route way.

**Comment by knowbody** [comment]
could you describe your use case then? when would you like to "refresh" your route?

**Comment by jaxxreal** [comment]
For example, I have several routes with complicated UI, something like step-by-step menus and forms in general. And when I'm using this functional, I can stop in the middle of UI-way, and want to start again. 

Yes, I can load initial state, but by my mind, do it for each similar route - wrong way.

**Comment by knowbody** [comment]
sorry not sure if I understand it correctly, so you want to reset your forms to their default state?

**Comment by jaxxreal** [comment]
Probably my English isn't ver well. I want route to get default state, but route has not only forms - it has sidebars and another menus. If I has this case for one route, I won't spam here. But I have many similar cases. When I used Angular, I triggered current location reload, and have no problem.

P.S. I found hacky way to do this trick. I remount ```RouteHandler``` by demand in app wrapper component, and all  happen by the way which I want.
